From: Mel Gorman <mgorman@suse.de>
Date: Tue, 2 Oct 2018 14:12:47 +0100
Subject: [PATCH] sched/numa: Pass destination CPU as a parameter to
 migrate_task_rq kabi

References: bnc#1101669 optimise numa balancing for fast migrate
Patch-mainline: No, queued in subsystem maintainer tree

This should be safe as no module should ever be accessing this opaque type
via task_struct.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 kernel/sched/sched.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 0ccac55fb5bf..2f40644d8de0 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -1399,7 +1399,11 @@ struct sched_class {
 
 #ifdef CONFIG_SMP
 	int  (*select_task_rq)(struct task_struct *p, int task_cpu, int sd_flag, int flags);
+#ifndef __GENKSYMS__
 	void (*migrate_task_rq)(struct task_struct *p, int new_cpu);
+#else
+	void (*migrate_task_rq)(struct task_struct *p);
+#endif
 
 	void (*task_woken) (struct rq *this_rq, struct task_struct *task);
 
